spotpy/algorithms/dds.py

```diff
@@ -0,0 +1,320 @@
+            * mpi: Message Passing Interface: Parallel computing on cluster pcs (recommended for unix os).
+
+        save_sim: boolean
+            * True:  Simulation results will be saved
+            * False: Simulation results will not be saved
+        """
+
+        super(DDS, self).__init__(*args, **kwargs)
+
+        self.np_random = np.random
+        self.min_bound, self.max_bound = self.parameter()['minbound'], self.parameter()['maxbound']
+
+        if hasattr(self.setup,"params"):
+            self.discrete_flag = [u.is_distinct for u in self.setup.params]
+        else:
+            self.discrete_flag = [False] * len(self.max_bound)
+
+    def _set_np_random(self, f_rand):
+        self.np_random = f_rand
+
+    def sample(self, repetitions, fraction1=0.2, trials=1, s_initial=[]):
+        """
+        Samples from the DDS Algorithm
+        :param repetitions:  Maximum number of runs.
+        :type repetitions: int
+        :param fraction1: value between 0 and 1
+        :type fraction1: float
+        :param trials: amount of runs DDS algorithm will be performed
+        :param s_initial: set an initial trial set
+        :return:
+        """
+
+        result_list = []
+        to_max = 1
+
+        self.set_repetiton(repetitions)
+
+        num_dec = len(self.min_bound)  # num_dec is the number of decision variables
+
+        if len(s_initial) == 0:
+            its = np.int(np.max([5, round(0.005 * repetitions)]))
+        elif len(s_initial) != num_dec:
+            raise ValueError("User specified 's_initial' has not the same length as available parameters")
+        else:
+            its = 1
+            s_initial = np.array(s_initial)
+            if not (np.all(s_initial <= self.max_bound) and np.all(s_initial >= self.min_bound)):
+                raise ValueError("User specified 's_initial' but the values are not within the parameter range")
+
+        for trial in range(trials):
+
+            # solution = np.array(repetitions * [(3 + num_dec) * [0.0]])
+            s_best = []
+            j_best = []
+
+            s_range = self.max_bound - self.min_bound
+
+            # =================================================================================================
+            # INITIAL SOLUTION
+            # =================================================================================================
+
+            if its > 1:  # its is the number of function evaluations to initialize the DDS algorithm solution
+                print('Finding best starting point for trial ' + str(trial + 1) + ' using ' + str(
+                    its) + ' random samples.')
+                i_left = repetitions - its  # use this to reduce number of fevals in DDS loop
+                if i_left <= 0:
+                    raise ValueError('# Initialization samples >= Max # function evaluations.')
+
+                starting_generator = (
```

Comment:
Use self.parameter() distribution function. Maybe we need to write a parameter.Discrete distribution